go/vt/vtgate/executor.go

```diff
@@ -1010,8 +1010,13 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 	_, _ = planHash.Write(hack.StringBytes(query))
 	planKey := hex.EncodeToString(planHash.Sum(nil))
 
-	if plan, ok := e.plans.Get(planKey); ok {
-		return plan.(*engine.Plan), nil
+	if qo.cachePlan() && sqlparser.CachePlan(statement) {
+		if plan, ok := e.plans.Get(planKey); ok {
+			if logStats != nil {
```

Comment:
I think we can skip doing nil check for `logStats`. It is expected to be not nil.

Reply:
I did it because on line 1002 there is a nil check as well.. If that is the case then I will remove that one as well? thoughts ?